frontend/src/metabase/actions/hooks/use-action-form/utils.ts

```diff
@@ -116,39 +115,27 @@ export const getInputType = (param: Parameter, field?: Field) => {
   return "string";
 };
 
-export const generateFieldSettingsFromParameters = (
-  params: Parameter[],
-  fields?: Field[],
-) => {
+export const generateFieldSettingsFromParameters = (params: Parameter[]) => {
   const fieldSettings: Record<ParameterId, LocalFieldSettings> = {};
 
-  const fieldMetadataMap = Object.fromEntries(
-    fields?.map(f => [slugify(f.name), f]) ?? [],
-  );
-
   params.forEach((param, index) => {
-    const field = fieldMetadataMap[param.id]
-      ? new Field(fieldMetadataMap[param.id])
-      : new Field({
-          id: param.id,
-          name: param.id,
-          slug: param.id,
-          display_name: humanize(param.id),
-          base_type: param.type,
-          semantic_type: param.type,
-        });
-
-    const name = param["display-name"] ?? param.name ?? param.id;
-    const displayName = field?.displayName?.() ?? name;
+    const field = new Field({
+      id: param.id,
+      name: param.id,
+      slug: param.id,
+      display_name: param["display-name"],
+      base_type: param.type,
+      semantic_type: param.type,
+    });
 
     fieldSettings[param.id] = getDefaultFieldSettings({
       id: param.id,
-      name,
-      title: displayName,
-      placeholder: displayName,
-      required: !!param?.required,
+      name: param.name,
+      title: param["display-name"],
```

Comment:
I think we prefer using methods like `field.displayName()` where we can rather than accessing the data directly. you may have found that this method didn't work because of the `display-name` vs `display_name` issue.

Reply:
Sounds good [Use field.displayName()](https://github.com/metabase/metabase/pull/30356/commits/ced6e5f00236c84bab551d5bccebe64b601e37be)